File: backend/aggregate_data.js
import Tesseract from 'tesseract.js';
import fs from 'fs';
import path from 'path';

async function aggregate(baseDir = 'book_data') {
    const outputFile = 'ai_knowledge_base.txt';
    let fullText = '';

    // Recursively find all PNGs in the book folders
    const getFiles = (dir) => {
        let results = [];
        const list = fs.readdirSync(dir);
        list.forEach(file => {
            const filePath = path.join(dir, file);
            const stat = fs.statSync(filePath);
            if (stat && stat.isDirectory()) results = results.concat(getFiles(filePath));
            else if (file.endsWith('.png')) results.push(filePath);
        });
        return results;
    };

    const images = getFiles(baseDir).sort();

    for (const img of images) {
        console.log(`📄 Reading: ${img}`);
        const { data: { text } } = await Tesseract.recognize(img, 'eng');
        fullText += `\n--- SOURCE: ${img} ---\n${text}\n`;
    }

    fs.writeFileSync(outputFile, fullText);
    console.log(`✅ Aggregation Complete! Data saved to ${outputFile}`);
}

aggregate();

File: backend/aggregate_data.py
import os
import pytesseract
from PIL import Image

def aggregate(base_dir="book_data"):
    output = "ai_knowledge_base.txt"
    with open(output, "w", encoding="utf-8") as f:
        for root, dirs, files in os.walk(base_dir):
            for file in sorted(files):
                if file.endswith(".png"):
                    path = os.path.join(root, file)
                    text = pytesseract.image_to_string(Image.open(path))
                    f.write(f"\n--- FILE: {path} ---\n{text}\n")
    print("Done! Data ready in ai_knowledge_base.txt")

if __name__ == "__main__":
    aggregate()
    /////////////